10562 redesign home page

roseeichelmann opened the pull request:
Fixes cityofaustin/atd-data-tech/issues/10562

This issue revamps our website's home page! One thing to note is that this PR does not address functionality/design for mobile devices/ or super small screens, so we will need to create another issue to address those UI considerations. I ended up addressing the hero section problem we talked about in dev sync on 10/31 and think I found a good solution. Open to any feedback on design and code!

Test steps:

[Netlify link](https://deploy-preview-50--atd-product.netlify.app/)

1. Mostly make sure all the links work and that the carousel for the featured issues functions as expected.
2. You should be able to click each carousel item and be rerouted to the page for that project or product.
3. Check all the tabs on the navbar and footer.

roseeichelmann commented on pages/index.js:
```diff
@@ -11,99 +14,125 @@ import IndexIsssueListItem from "../components/shared/IndexIssueListItem";
+
+  issues.map((issue) => {
+    if (issue.isFeatured) {
+      featuredIssues.push(issue);
```
We used to use only projects for the featured tab, now we use both products and projects so I import all issues and create a featuredIssues array based on the label.

roseeichelmann commented on pages/index.js:
```diff
@@ -11,99 +14,125 @@ import IndexIsssueListItem from "../components/shared/IndexIssueListItem";
+              <Carousel.Item>
+                <Row>
+                  <Col sm={1}></Col>
+                  {featuredIssues.slice(0, 4).map((issue) => {
```
This doesn't feel super elegant and the code is a little redundant. It also requires us to only have 8 featured issues so its not very reusable if we want more or less featured issues in the future. But since we want 4 cards on each "slide" of the carousel and 8 total cards, this was the solution I came up with. Let me know if you have other ideas!

roseeichelmann replied:
This is a good point, I'll definitely use your idea to refactor the carousel into it's own component. Thanks Mike!

roseeichelmann replied:
Also yeah, the way the cards are rendered in the carousel is the same way they are rendered in the [Projects](https://deploy-preview-50--atd-product.netlify.app/projects?status=in_progress) and [Products](https://deploy-preview-50--atd-product.netlify.app/products) pages, using the same `IndexIsssueListItem` component. So if I change the design for the carousel cards it will also change the cards on the other pages. @patrickm02L would you rather keep the design as is or change the image to being on the top half of the card which would also change how they look in those other pages?

patrickm02L replied:
Please change the design of the cards so they mirror what's rendered in the Data & Performance hub. 
https://data.mobility.austin.gov/

Image
Title
Description

chiaberry replied:
what should happen for the cards that don't have images?

roseeichelmann replied:
@patrickm02L Just to double check, should this change also happen for the cards on the Products and Projects pages?

patrickm02L replied:
@roseeichelmann yes ultimately all the cards will have the new design.

mddilley commented on pages/index.js:
```diff
@@ -11,99 +14,125 @@ import IndexIsssueListItem from "../components/shared/IndexIssueListItem";
-              alt="Illustration of a green bicycle"
-            />
-          </Col>
+          <Card className="hero-section p-0 h-100">
+            <Card.Body className="pt-5">
+              <Card.Title className="pt-4 text-center text-white fw-bold">
+                <h1 className="fw-bold" style={{ fontSize: "3.5rem" }}>
+                  Tech Support for ATX Mobility
+                </h1>
+              </Card.Title>
+              <Card.Text className="pt-3 text-center text-white">
+                <p style={{ fontSize: "1.75rem", fontWeight: 500 }}>
+                  We build and buy technology to help city staff <br></br>
+                  tackle Austin&apos;s mobility challenges.
+                </p>
+              </Card.Text>
+            </Card.Body>
+          </Card>
```
Taking a quick pass through this - I think that these card components could be replaced with elements similar to the first example in [this Bootstrap 5 hero section](https://getbootstrap.com/docs/5.0/examples/heroes/) that also use [display headings](https://getbootstrap.com/docs/5.0/content/typography/#display-headings) for font size and uses elements/classes that don't have a border-radius OOTB.  

That way you don't have to work around all the built-in opinions of the Card components.

roseeichelmann replied:
TIL about display headings! Will definitely look into these. Thanks!

mddilley reviewed (commented):
This is looking super good! Left some feedback to expand beyond the dev sync discussion yesterday.

roseeichelmann commented:
I also just realized theres an extra Moped v1.0 card hanging around in the carousel section that shouldn't be there. It will be gone tomorrow once the data from github is updated tonight, so ignore that!

chiaberry commented on components/shared/NavComponent.js:
```diff
@@ -87,24 +89,24 @@ export default function NavComponent(props) {
             })}
             <a
               href="https://atd.knack.com/dts#new-service-request"
+              //why are there two of these?
```
one is hidden at d-md-none and the other one is hidden at d-none

patrickm02L reviewed (commented):
**Changes**

:18 - Unbold the Nav "Product, Projects, About" buttons
:26 - Take the curved edges off the hero image
:42 - Update the cards so that they mirror the Data & Performance hub set up (image, title, description)
1:29 - Four carousel cards per set
1:42 - Carousel arrows should nots show when cards slide through

https://user-images.githubusercontent.com/5566727/199308551-a6b8ad63-de1c-48d7-97cb-65b84021d37e.mp4

roseeichelmann commented:
I'm having trouble getting the cards in the featured carousel to look cohesive because some of them have really long descriptions and all the photos are different sizes.

This is what the cards look like if I leave the images their original sizes:

![Screenshot from 2022-11-01 16-41-47](https://user-images.githubusercontent.com/67130963/199347785-69b78bcd-6d63-4d4d-bec6-b0fc0f95a033.png)

Cards with fixed image size but some descriptions are just a lot longer which may be off putting:

![Screenshot from 2022-11-01 16-54-16](https://user-images.githubusercontent.com/67130963/199349326-158562e7-3d46-4d9a-a6f7-3f42b2b4bcdf.png)

Solutions I have are to either make the descriptions shorter like they are in https://data.mobility.austin.gov/ pr completely nix them so we just have the title like this:

![Screenshot from 2022-11-01 16-44-12](https://user-images.githubusercontent.com/67130963/199348237-38ca8a2f-618f-4c7a-88bc-c7d8f6f77378.png)

Thoughts?  @patrickm02L

roseeichelmann commented on pages/index.js:
```diff
@@ -11,99 +14,125 @@ import IndexIsssueListItem from "../components/shared/IndexIssueListItem";
+    if (issue.isFeatured) {
+      featuredIssues.push(issue);
+    }
+  });
```
Done, this is so much cleaner!

frankhereford reviewed (approved):
This is really really really ridiculously good looking! 
![zoolander-bluesteel](https://user-images.githubusercontent.com/3653206/199508021-32767f08-5ac4-454c-8187-ad1e2fde061f.gif)

Thank you for your work updating this. The quality of our website reflects tremendously on our team, and your work here represents us so well!

johnclary commented on components/FeaturedCarousel.js:
```diff
@@ -0,0 +1,54 @@
+import { useContext } from "react";
+import IssuesContext from "../contexts/IssuesContext";
+import IndexIsssueListItem from "../components/shared/IndexIssueListItem";
+import { Row, Col, Carousel } from "react-bootstrap";
```
i think nextjs does a good job of optimizing this for you, but you should import each component using its default export. There's a lil blurb about it in the docs, [here](https://react-bootstrap.github.io/getting-started/introduction/#importing-components).

```javascript
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Carousel";
import Carousel from "react-bootstrap/Carousel";
```

johnclary commented on components/FeaturedCarousel.js:
```diff
@@ -0,0 +1,54 @@
+import { useContext } from "react";
+import IssuesContext from "../contexts/IssuesContext";
+import IndexIsssueListItem from "../components/shared/IndexIssueListItem";
```
you have an extra `s` in this import name `IndexIsssueListItem`

johnclary commented on components/FeaturedCarousel.js:
```diff
@@ -0,0 +1,54 @@
+      <Carousel.Item>
+        <Row className="mb-3">
+          <Col sm={1}></Col>
+          {featuredIssues.slice(4, 8).map((issue) => {
```
Four cards are not rendering well on mobile. Assuming we'll want to tweak the number of cards per page on mobile, you might assign the size of each slice to a variable. something like:

```javascript
featuredIssues.slice(CARDS_PER_PAGE, CARDS_PER_PAGE * 2).map((issue) => {
```

johnclary replied:
It also might be worth adding a note here that y'all have decided that at most 8 featured projects will ever be displayed?

johnclary commented on components/shared/NavComponent.js:
```diff
@@ -11,9 +11,9 @@ import {
-  { label: "About us", route: "/about" },
+  { label: "Products", route: "/products" },
+  { label: "Projects", route: "/projects" },
+  { label: "About", route: "/about" },
```
much better!

johnclary commented on pages/index.js:
```diff
@@ -1,109 +1,77 @@
+              We build and buy technology to help city staff <br></br>
+              tackle Austin&apos;s mobility challenges.
+            </h3>
+          </div>
```
I like the shorter copy here. It's a bit hard to read the `<h3>` white text against the background. maybe a bit of text shadow could help?

roseeichelmann replied:
I just added some text shadows to both the main hero title and this `h3` let me know what you think!

johnclary commented on pages/index.js:
```diff
@@ -1,109 +1,77 @@
-            <h1 className="homepage-header">
-              <strong>Tech support for ATX mobility</strong>
+        <Row className="hero-section d-flex align-items-center">
+          <div>
```
Glad yo see you moved away from the `Card` component. This is great. 

Any reason not to use `<Col>` instead of a plain `<div>` here?

roseeichelmann replied:
Absolutely no good reason just an oversight on my part. Fixed it!

johnclary commented on pages/index.js:
```diff
@@ -1,109 +1,77 @@
-            {/* hidden on xs and sm */}
-            <Image
-              fluid
-              src="/assets/sports.jpg"
```
@roseeichelmann you can safely delete any assets like this that we're not longer using.

johnclary reviewed (changes requested):
This looks sooo good! I had just a few minor changes. I can't wait for this to ship! 🙌

patrickm02L commented:
> I'm having trouble getting the cards in the featured carousel to look cohesive because some of them have really long descriptions and all the photos are different sizes.
>
> Thoughts? @patrickm02L

Opened this up to the team in Slack, got lots of good [feedback](https://austininnovation.slack.com/archives/CHZE6BC6L/p1667393345824789) and landed on truncating the description so the cards keep their uniformity.

roseeichelmann commented:
Just pushed some updates to this PR! Thing I changed:

- Text shadows for the hero section for better readability

- All of the cards and card images are now a fixed size

- Next/prev controls disappear while slides are transitioning

- Changed the card carousel descriptions to include just the first sentence of issue instead of first line break unless the first sentence doesn't have a period (some issues are like this) in which case first line break is used.

- Added an ellipses truncation after 3 lines of text in the card descriptions

roseeichelmann commented on components/FeaturedCarousel.js:
```diff
@@ -0,0 +1,63 @@
+      interval={null}
+      controls={showArrows}
+      onSlide={(e) => setShowArrows(false)}
+      onSlid={(e) => setShowArrows(true)}
```
this makes the arrows hide while sliding

roseeichelmann commented on components/shared/IndexIssueListItem.js:
```diff
@@ -1,14 +1,15 @@
+  const reDescriptionLinebreak = /.*/;
+  const description =
+    body.match(reDescriptionSentence)?.[0] ||
+    body.match(reDescriptionLinebreak)[0];
```
use the first sentence as a description unless there is no period in which use first linebreak

roseeichelmann commented on styles/custom.scss:
```diff
@@ -71,6 +67,37 @@ $hr-opacity: 1;
+.card-text {
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 3;
```
ellipses truncation after 3 lines 🤝 side note - is using webkit for this okay? i see we use it elsewhere in VZE and moped

frankhereford replied:
@roseeichelmann wrote:
> is using webkit for this okay?

I say 100% yes. Chrome + safari are webkit, so that's a real good chunk of browsers right there. I'd say, if it works for you on Firefox on ubuntu, then it's going to almost always going to be a supported feature.

👍

roseeichelmann replied:
It does indeed work on my ubuntu + firefox setup, so I'm confident it should work for most! @frankhereford

frankhereford commented on pages/index.js:
```diff
@@ -1,109 +1,78 @@
+        <Row className="hero-section d-flex align-items-center">
+          <Col>
+            <h1 className="text-white display-3 fw-bold text-center pb-4">
+              Tech Support for ATX Mobility
```
Short words are good words in a hero image, but I get the heebie jeebies saying "tech support." is it just me? IT Support? Information Technology Support? Technology Support?

frankhereford commented on pages/index.js:
```diff
@@ -1,109 +1,78 @@
+        <Row className="hero-section d-flex align-items-center">
+          <Col>
+            <h1 className="text-white display-3 fw-bold text-center pb-4">
+              Tech Support for ATX Mobility
```
Short words are good words in a hero image, but I get the heebie jeebies saying "tech support." is it just me? IT Support? Information Technology Support? Technology Support?

roseeichelmann replied:
It doesn't bother me personally!

frankhereford reviewed (approved):
Thank you for the refinements - I can't wait to see this one on the way out of the harbor! 🚢 🚢 🚢 ✨

patrickm02L reviewed (approved):
This is SO clean! Very well done. 🤘
https://user-images.githubusercontent.com/5566727/199715958-4cc1e6f0-2691-47cf-9049-d0c84e765464.mp4

tillyw reviewed (approved):
this looks amazing!! i love the solution you implemented for the carousel image size and description length discrepancies, it all looks really uniform now that the descriptions are truncated after 3 lines :sparkles: all the links work and everything functions as expected. excited to see this merged and to see the mobile enhancements that are coming!

mddilley reviewed (approved):
✨ ✨ ✨ 🚢

roseeichelmann commented on pages/index.js:
```diff
@@ -1,109 +1,78 @@
-            ) : null;
-          })}
+          <Col>
+            <FeaturedCarousel></FeaturedCarousel>
```
This is a question I've had before, when to use a self-closing tag, and you've just answered it. Thanks!
